### PR TITLE
test: add integration test with Canal CNI and reset API

### PR DESCRIPTION
**Pull request by smira**
Canal CNI is known to be trying to reach out to k8s control plane on pod
teardown.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Comment by smira** [comment]
e2e still fails, need to figure out why

File: .drone.jsonnet [hunk 1]
```diff
@@ -314,6 +321,7 @@ local default_steps = [
   e2e_iso,
   e2e_qemu,
   e2e_docker,
+  xxx_integration_canal_reset, // TODO: just to test it out
```
Comment (smira) [inline .drone.jsonnet:324]: TODO: remove me

**Review by rsmitty — APPROVED** [review]

**Review by andrewrynhard — APPROVED** [review]

**Pull request merged**